### ISSUE TITLE
JSON RPC Player.Stop() seems not to close rtsp tcp command port

### ISSUE THREAD

**Issue by stko** [xbmc/xbmc#21280]
### Describe the bug
When closing a rtsp sat->IP receiver stream with Player.Stop() and opening a new channel with Player.Open(rtsp://..), it always fails if I do not wait approc. 60 secs between Stop() and Open()

Configuration: A SAT->IP Server providing RTSP-TCP Streams, where the channel is defined by a
```
rtsp://192.168.1.113/?freq=11493&src=aa&pol=h&msys=dvbs2&sr=22000&fec=23&rolloff
=35&sid=10301&pmt=5100&pids=0,5101,5102,5100
``` 
-URL

This channel can be played by JSON RPC Player.Open(URL).

When then stopping the channel with Player.Stop() and start a new one with Player.Open() again, I get an error message on the screen and an entry in the log (see further below)

Then I've to wait for around 60secs to try again before it works. From time to time even the sat receiver hangs up on that and need to be restarted.

But when not using the Stop() command in between two Open() commands, I'm able to switch the channels instandly by the Open() command.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

I would expect that a Stop() between two Open() should not have any negative effect.

## Actual Behavior

using Stop() on RTSP causes a 60 sec. breakdown of the sat receiver communication and sometimes a complete hang up.

## Possible Fix

As investigated during the [RTSP implementation](https://forum.kodi.tv/showthread.php?tid=359072&pid=3094863#pid3094863), RTSP uses a tcp channel to send commands to the sat receicer. It appears as if kodi does not send a TEARDOWN command to terminate the stream sending, it probably just interrupts the tcp channel, which leaves the receicer in a 60 secs connection interruption timeout, waiting for the connection to come back.

When using two open() commands in sequence instead, then maybe the tcp connection is not interrupted and re-used for the second open-stream command. This would explain why two open() commands do work.

So the question would be, if the Stop() command does a rtsp compliant connection shutdown or not..?

### To Reproduce

As descripted above

### Debuglog

The debuglog can be found here:
```
2022-04-18 12:25:04.370 T:583      INFO <general>: VideoPlayer::OpenFile: rtsp://192.168.1.113/?freq=11493&src=aa&pol=h&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=10301&pmt=5100&pids=0,5101,5102,5100
2022-04-18 12:25:04.374 T:12619    INFO <general>: Creating InputStream
2022-04-18 12:25:04.378 T:12619    INFO <general>: Creating Demuxer
2022-04-18 12:25:04.418 T:12619   ERROR <general>: OpenDemuxStream - Error creating demuxer
2022-04-18 12:25:04.418 T:12619    INFO <general>: CVideoPlayer::OnExit()
2022-04-18 12:25:04.522 T:583      INFO <general>: CVideoPlayer::CloseFile()
2022-04-18 12:25:04.522 T:583      INFO <general>: VideoPlayer: waiting for threads to exit
2022-04-18 12:25:04.522 T:583      INFO <general>: VideoPlayer: finished waiting

```

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

LibreElec w. Kodi 19
Raspberry 3
Sat Receiver: Telestar Digitbit R1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by a1rwulf** [comment]
Please add a full debug log.

**Comment by stko** [comment]
I've discussed that item also on https://forum.kodi.tv/showthread.php?tid=359072&pid=3095568#pid3095568 with @phunkyfish 

As it seems, the issue is driven by the circumstance that the wrong protocol rtsp:// instead satip:// is used, and this came from that satip:// was erroneously not included in the list of supported protocols.

```
2022-04-24 06:46:12.432 T:2123  WARNING <general>: Create - unsupported protocol(satip) in satip://192.168.1.113/?freq=11347&src=aa&pol=v&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=11170&pmt=6700&pids=0,6710,6720,6700
2022-04-24 06:46:12.433 T:2123    ERROR <general>: Exists - Error checking for satip://192.168.1.113/?freq=11347&src=aa&pol=v&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=11170&pmt=6700&pids=0,6710,6720,6700
2022-04-24 06:46:12.490 T:584      INFO <general>: VideoPlayer::OpenFile: satip://192.168.1.113/?freq=11347&src=aa&pol=v&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=11170&pmt=6700&pids=0,6710,6720,6700
2022-04-24 06:46:12.492 T:2126     INFO <general>: Creating InputStream
2022-04-24 06:46:12.494 T:2126  WARNING <general>: CreateLoader - unsupported protocol(satip) in satip://192.168.1.113/?freq=11347&src=aa&pol=v&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=11170&pmt=6700&pids=0,6710,6720,6700
2022-04-24 06:46:12.494 T:2126    ERROR <general>: CVideoPlayer::OpenInputStream - error opening [satip://192.168.1.113/?freq=11347&src=aa&pol=v&msys=dvbs2&sr=22000&fec=23&rolloff=35&sid=11170&pmt=6700&pids=0,6710,6720,6700]
2022-04-24 06:46:12.494 T:2126     INFO <general>: CVideoPlayer::OnExit()

```

@phunkyfish has already provided a patch for that (https://github.com/xbmc/xbmc/pull/21281) and I'm just working to build a LibreElec test image for that, so more info will hopefully come during the day

**Comment by stko** [comment]
I've just tested the patch, and the original tcp reconnect timeout does obviously not appear anymore. I'll do further testing, but for now everything seems to be ok

Closed by #21281
